Signal/src/util/DateUtil.m

```diff
@@ -259,14 +259,20 @@ + (NSString *)formatTimestampShort:(uint64_t)timestamp
 
 + (NSString *)formatDateShort:(NSDate *)date
 {
+    OWSAssertIsOnMainThread();
     OWSAssert(date);
 
+    NSDate *now = [NSDate date];
+    NSInteger dayDifference = [self daysFromFirstDate:date toSecondDate:now];
+    BOOL dateIsOlderThanToday = dayDifference > 0;
+    BOOL dateIsOlderThanOneWeek = dayDifference > 6;
```

Comment:
Date math is expensive; this cuts the cost in half.

Reply:
Hmm... any insight into why it's expensive?

It looks like this change would approximately cut it in half?

Reply:
It's the usage of `NSCalendar` to slice the date into components that is surprisingly expensive. Due to the relative (to now) nature of these methods, we can't cache.